### PR TITLE
Create action to build binaries

### PR DESCRIPTION
**Pull request by smaye81**
This adds a Github Action to build the conformance runner binaries for Linux and MacOS for x86_64 architectures. Building for ARM64 is planned for a follow-up as it is not very easy to build these using Github runners. Github does not provide native ARM64 images for Linux or Mac.

The trigger for the action is a tag being pushed to the repo. The plan is for the tags in this repo to mirror the tags from Protobuf. Ideally, it would be nice to poll the [Protobuf repo](https://github.com/protocolbuffers/protobuf) for releases, but that isn't an easy task either and could be done in a follow-up also. 

Pushing an invalid tag not found on the Protobuf repo will cause the action to fail.

Example of successful run (note: release and tags were deleted after the run):  https://github.com/bufbuild/protobuf-conformance/actions/runs/4692167405

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -46,6 +46,17 @@ test: $(BIN)/conformance_test_runner  ## Run conformance tests
 	cd impl/baseline;        PATH="$(abspath $(BIN)):$(PATH)" ./test.sh
 	node report.js
 
+.PHONY: testci
```
[the remark this answers is not in the document]
Comment (timostamm) [reply Makefile:49]: No idea, this is not terrible IMO.

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+      - name: Run Conformance Tests
+        run: |
+          make testci
+
+      - name: Publish GitHub artifacts
+        uses: softprops/action-gh-release@v1
+        with:
+          generate_release_notes: false
+          body: Conformance runner binaries for Protobuf v${{ needs.get-protobuf.outputs.version }}
```
[the remark this answers is not in the document]
Comment (smaye81) [reply .github/workflows/release.yaml:85]: Yep, sorry, oversight on my part.

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+          mkdir -p .tmp/bin
+          cp ./protobuf-${{ needs.get-protobuf.outputs.version }}/bazel-bin/conformance/conformance_test_runner .tmp/bin/conformance_test_runner-${{ needs.get-protobuf.outputs.version }}-osx-x86_64
+
+      - name: Publish GitHub artifacts
+        uses: softprops/action-gh-release@v1
+        with:
+          generate_release_notes: true
+          append_body: true
+          files: |
+            .tmp/bin/conformance_test_runner-${{ needs.get-protobuf.outputs.version }}-osx-x86_64
```
[the remark this answers is not in the document]
Comment (smaye81) [reply .github/workflows/release.yaml]: Good call. Updated to zip up the binaries.

**Review by timostamm — APPROVED** [review]
Looks great to me! We might want to follow up with a short doc about the runners, and how to use them along with the proto files.

**Pull request merged**